packages/hoppscotch-common/src/components/http/Sidebar.vue

```diff
@@ -33,6 +33,24 @@
     >
       <Share />
     </HoppSmartTab>
+    <HoppSmartTab
+      :id="'codegen'"
+      :icon="IconCode"
+      :label="`${t('tab.codegen')}`"
+    >
+      <div
+        class="flex items-center overflow-x-auto whitespace-nowrap border-b border-dividerLight px-4 py-2 text-tiny text-secondaryLight"
+      >
+        <span class="truncate"> Request </span>
+        <icon-lucide-chevron-right class="mx-2" />
+        Code snippet
+      </div>
+      <HttpCodegen
+        :hide-label="true"
+        class="px-4 mt-4"
+        v-if="selectedNavigationTab === 'codegen'"
```

Comment:
why do we need the v-if here ?

Reply:
Because we only want to process codegen when the tab is open.

Reply:
okay gotcha